src/connectiontcpserver.cpp: fix musl build

Build on musl fails on:

connectiontcpserver.cpp:131:13: error: 'strerror' was not declared in this scope
           + strerror( err ) + " (errno: " + util::int2string( err ) + ")";
             ^~~~~~~~

Fixes:
 - http://autobuild.buildroot.org/results/a61a91bb6e1156eff5fb756f9dccfc2bb6c0f277

Signed-off-by: Fabrice Fontaine <fontaine.fabrice@gmail.com>
[Upstream status: https://bugs.camaya.net/ticket/?id=291]

diff -Nuar gloox-1.0.23-orig/src/connectiontcpserver.cpp gloox-1.0.23/src/connectiontcpserver.cpp
--- gloox-1.0.23-orig/src/connectiontcpserver.cpp	2020-06-05 23:22:59.624018949 +0200
+++ gloox-1.0.23/src/connectiontcpserver.cpp	2020-06-05 23:26:20.708027221 +0200
@@ -36,6 +36,7 @@
 # include <resolv.h>
 # include <netdb.h>
 # include <arpa/inet.h>
+# include <string.h>
 # include <sys/socket.h>
 # include <sys/un.h>
 # include <sys/select.h>
